lib/rex/proto/proxy/socks5.rb

```diff
@@ -0,0 +1,543 @@
+
+# -*- coding: binary -*-
+#
+# sf - Sept 2010
+# surefire - April 2018
+#
+# TODO: Support SOCKS authentication (just accept anything, I guess)
+# TODO: IPv6 support
+# TODO: Hostname lookups.  Do they work?
```

Comment:
Presumably we want to do hostname lookups through the session. This may require some rework at the parameter validation level to prevent the current normalization that occurs. I'd say it's out of scope here probably.